spec/section_spec.rb

```diff
@@ -68,9 +68,11 @@ class Page < SitePrism::Page
     expect(SitePrism::Section).to respond_to :elements
   end
 
-  it 'passes a given block to Capybara.within' do
-    expect(Capybara).to receive(:within).with('div')
-    SitePrism::Section.new(a_page, 'div') { 1 + 1 }
+  it 'passes a given block to #within' do
+    block_variable = 5
+    expect_any_instance_of(SitePrism::Section).to receive(:within).and_yield
```

Comment:
use of `expect_any_instance` is discouraged. Also, this is stubbing the subject of the test.

The original expectation on `Capybara` should probably be kept but updated to use `and_yield` as you are doing here.